src/coreclr/gc/satori/SatoriRecycler.cpp

```diff
@@ -896,6 +896,8 @@ void SatoriRecycler::BlockingMarkForConcurrent()
 
         size_t blockingDuration = (GCToOSInterface::QueryPerformanceCounter() - blockingStart);
         m_CurrentGcInfo->m_pauseDurations[1] = blockingDuration / m_perfCounterTicksPerMicro;
+        m_gcDurationMillis[0] += blockingDuration / m_perfCounterTicksPerMicro;
```

Comment:
This should be `[m_condemnedGeneration]`  it can be either 1 or 2 here.

Reply:
Also - `m_gcDurationMillis` is used in `GetLastGCDuration`. That reports the duration of just the last GC.

For the accumulating pause counter we need another set of counters. I think we only need to track gen1 and gen2 though. 

That would be consistent with not reporting gen0 in `GetLastGCDuration`.

Reply:
The "pause" APIs are questionable for gen0 collections that are per-thread and do not affect the entire generation, do not stop the app, and also several may happen at the same time. 
I.E. - should the pause time be added if pauses overlap?

In `GetLastGCDuration` we could report the duration of the last which completed, but it would imply that every Gen0 collection stores its duration into a static field. That may have some cost due to contentions, while the value of this metric feels questionable.

I think it is better to not involve gen0 in all this pause/duration deal - at least until there is a more clear reason for how that would be useful and better understanding of how to report it.